File: 1-two-sum/two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        
        HashMap<Integer,Integer> numAndInd = new HashMap<>();
        int ans[] = new int[2];
        
        for(int i =0;i<nums.length;i++)
        {
            int numToFind = target - nums[i];
            if(numAndInd.containsKey(numToFind))
            {
                ans[0] = i;
                ans[1] = numAndInd.get(numToFind);
            }
            numAndInd.put(nums[i],i);
        }
        
        return ans;
    }
}

File: 1-two-sum/two-sum.go
func twoSum(nums []int, target int) []int {
    hash := make(map[int]int)
    n := len(nums);
    result := make([]int,2);
    for i := 0;i<n;i++ {
        rem := target - nums[i]
        if ind,ok := hash[rem]; ok {
            result[0] = ind
            result[1] = i
            break;
        }
        hash[nums[i]] = i
    }
    return result
}